### PR TITLE
Use CubinLinker for CUDA Minor Version Compatibility

### PR DESCRIPTION
**Pull request by gmarkall**
## Description

This switches to using CubinLinker (from PTXCompiler, but CubinLinker uses PTXCompiler internally) for Minor Version Compatibility. This enables support for all Numba features except linking archives with MVC, in support of use cases such as String UDFs (#11319) with MVC.

## Checklist
- [X] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [X] New or existing tests cover these changes.
- [X] The documentation is up to date with these changes.

### CODE REVIEW

File: conda/environments/cudf_dev_cuda11.5.yml [hunk 1]
```diff
@@ -7,6 +7,7 @@ channels:
   - rapidsai-nightly
   - dask/label/dev
   - conda-forge
+  - gmarkall
```
Comment (galipremsagar) [inline conda/environments/cudf_dev_cuda11.5.yml:10]: You will need to have the channel added into `build.sh` files in the similar way I did in my PR here: https://github.com/rapidsai/cudf/pull/11617/files

Comment (gmarkall) [reply conda/environments/cudf_dev_cuda11.5.yml:10]: Thanks, I've just seen this whilst I was making a change with a similar aim. I will have a look at the way you did it too.

**Comment by galipremsagar** [comment]
CI errors related to :https://github.com/dask/dask/issues/9490

File: python/cudf/cudf/core/udf/__init__.py [hunk 2]
```diff
@@ -28,36 +24,32 @@
...
-        del strings_udf
-
+    from strings_udf import ptxpath
+    from strings_udf._lib.cudf_jit_udf import to_string_view_array
+    from strings_udf._typing import str_view_arg_handler, string_view
+
+    from . import strings_typing  # isort: skip
+    from . import strings_lowering  # isort: skip
+
+    # add an overload of MaskedType.__init__(string_view, bool)
```
Comment (brandon-b-miller) [inline python/cudf/cudf/core/udf/__init__.py:34]:
```suggestion
```
Removing this comment should fix the isort/black conflict here.

**Comment by gmarkall** [comment]
Now that all the required packages are available in the usual channels (Numba 0.56.2 from conda-forge and CubinLinker from rapidsai) I've removed the "DO NOT MERGE" label. Additionally I've added some changes from Brandon that remove the checks on the driver / runtime versions for enabling string UDFs, as the changes in this branch mean that those checks should no longer be necessary.

Assuming the tests all pass this should be good for a review / merge if the review is OK.

**Comment by galipremsagar** [comment]
I opened an integration repo PR: https://github.com/rapidsai/integration/pull/535

**Review by raydouglass — APPROVED** [review]

File: ci/gpu/build.sh [hunk 3]
```diff
@@ -266,27 +266,14 @@ py.test -n 8 --cache-clear --basetemp="$WORKSPACE/custreamz-cuda-tmp" --junitxml
...
-    exit ${STRINGS_UDF_PYTEST_RETCODE}
-else
-    cd "$WORKSPACE/python/cudf/cudf"
-    gpuci_logger "Python py.test retest cuDF UDFs"
-    py.test tests/test_udf_masked_ops.py -n 8 --cache-clear
-fi
+# retest cudf with strings_udf present
+py.test -n 8 --cache-clear --basetemp="$WORKSPACE/strings-udf-cuda-tmp" --junitxml="$WORKSPACE/junit-strings-udf.xml" -v --cov-config=.coveragerc --cov=strings_udf --cov-report=xml:"$WORKSPACE/python/strings_udf/strings-udf-coverage.xml" --cov-report term tests
+gpuci_logger "Python py.test retest cuDF UDFs"
+py.test tests/test_udf_masked_ops.py -n 8 --cache-clear
```
Comment (shwina) [inline ci/gpu/build.sh:276]: Is this right? Don't we need to `cd $WORKSPACE/python/cudf/cudf` between these two lines?

Comment (brandon-b-miller) [reply ci/gpu/build.sh:276]: you're right. Looks like I was a little overzealous with my deletions here @gmarkall , sorry about that!

**Comment by jakirkham** [comment]
Thanks all! 🎉

**Pull request merged**